lib/prompt/parent_branches.go

```diff
@@ -33,5 +46,70 @@ func askForBranchAncestry(branchName string) {
 }
 
 func askForParentBranch(branchName string) string {
-	panic("unimplemented")
+	branchNameFmt := color.New(color.Bold).Add(color.FgCyan)
+	mainBranch := config.GetMainBranch()
+	message := fmt.Sprintf("Please specify the parent branch of %s by name or number (default: %s): ", branchNameFmt.Sprintf(branchName), mainBranch)
+	numberedBranches := git.GetLocalBranchesWithMainBranchFirst()
+	numericRegex, err := regexp.Compile("^[0-9]+$")
+	if err != nil {
+		log.Fatal("Error compiling numeric regular expression: ", err)
+	}
+
+	for {
+		fmt.Printf(message)
+		userInput := util.GetUserInput()
+		parent := ""
+		if numericRegex.MatchString(userInput) {
```

Comment:
Haha, its far shorter than the one in bash, but yes still quite long. I'll try to extract the different parts. Parsing the user input could go into its own function (which could possibly be decomposed into other functions as well)